src/lib/operators/join_hash.cpp

```diff
@@ -281,16 +263,32 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
     jobs.back()->schedule();
 
     jobs.emplace_back(std::make_shared<JobTask>([&]() {
-      // Materialize right table. 'keep_nulls' makes sure that the relation on
-      // the right materializes NULL values when executing an OUTER join.
-      materialized_right = materialize_input<RightType, HashedType>(right_in_table, _column_ids.second,
-                                                                    histograms_right, _radix_bits, keep_nulls);
+      // Materialize right table. The third template parameter signals if the relation on the right (probe
+      // relation) materializes NULL values when executing OUTER joins (default is to discard NULL values).
+      /*
+      TODO (anyone):
+        Clang 6.0.1 does not allow "materialize_input<RightType, HashedType, static_cast<bool>(keep_nulls)>" (error:
+        invalid explicitly-specified argument), which works on GCC (see https://stackoverflow.com/q/33872039/1147726).
```

Comment:
Wait, what? Why does gcc accept that? I thought template arguments would have to be constexpr. `keep_nulls` obviously is not. Generally, it would be news to me if that would work: https://godbolt.org/z/xTZ6jc

The difference to the SO post is that they are using a constexpr expression as the template parameter.

If you want to, try and isolate a piece of code with a runtime value that gcc accepts but clang doesn't.

Reply:
Why should they have to be constexpr? 

Do we talk about the same? https://godbolt.org/z/1e9R8w

Reply:
At least known at compile-time.

```
int main(int argc, char **argv) {
    const bool x = argc > 1;
    foo<x>();
}
```
does not work

Reply:
Ah ok, I see.

So simply remove the comment or is there an alternative to implement?

Reply:
Not that I am aware of. I'd love it if it would work the way you're suggesting, but I don't think the standard supports it

Reply:
read: remove the comment